src/components/ListToolbox/ListToolbox.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import { bem } from '../../utils';
+import { Button } from '../Buttons/Button';
+import styles from './ListToolbox.scss';
+
+type Selection = {
+    amount: number;
+    label: string;
+};
+
+type Toggle = {
+    label: string;
+    onClick: () => void;
+};
+
+type Toggles = {
+    current: 'all' | 'none';
+    all: Toggle;
+    none: Toggle;
+};
```

Comment:
I am thinking about how to make `Toggles` interface more simple.
What do you think if we introduce new props:
```
isAllItemsSelected: boolean
allToggle: Toggle
noneToggle: Toggle
```

Reply:
In terms of selection, we can have 3 situations (let's say we have 10 items per page).
And if we have a look at GMail as a point of reference, the behaviour is the following:

| Selected items  | Toggle state |
| --------------- | ------------- |
| 0 / 10  | Select All |
| 1–9 / 10  | Select None  |
| 10 / 10  | Select None  |

So, `isAllItemsSelected` (which is 10/10) doesn't solve the problem. It actually should be something like `isNothingSelected`. I'd rather keep it as `currentToggleState = "all" | "none"` which is more straightforward and easy to reason about as for me.

Although, I don't like how current `toggles` property looks like as well. It makes an impression that there can be more then 2 states. But I still like the idea that all the props related to the toggle are grouped together.

How about this?

We can have `currentToggleState: "all" | "none"` property and `toggle: Toggle` property?

```ts
type Toggle = {
  selectAllLabel: 'string';
  selectNoneLabel: 'string';
  onToggle: (toggleState: "all" | "none") => void;
};
```



`isAllItemsSelected` sounds bit confusing, since in the situation when`All Items Selected`

Reply:
yep, it looks better. In this scenario, the `toggleState` type should be exported as well.